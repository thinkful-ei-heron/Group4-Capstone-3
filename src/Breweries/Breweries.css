.breweries-list {
	width: 30%;
	height: 75vh;
	overflow-y: scroll;
	background-color: white;
	z-index: 1;
}
.breweries-page {
	background-image: url('../images/wall6.jpg');
}

@media only screen and (max-width: 570px) {
	.breweries-page input:checked ~ a {
		margin-bottom: 0px;
	}

	.breweries-page nav {
		background-color: transparent;
	}
	.breweries-page .header-component a {
		font: 10px;
		margin: 0px;
	}
	.breweries-page nav input {
		display: inline-block;
		width: 60px;
		height: 50px;
		position: absolute;
		top: 123px;
		left: 2px;
		cursor: pointer;
		opacity: 0;
		-webkit-touch-callout: none;
	}
	.breweries-page nav span {
		display: block;
		width: 33px;
		height: 4px;
		margin-bottom: 5px;
		position: relative;
		background: white;
		border-radius: 3px;
		z-index: 1;
		transform-origin: 4px 0px;
		transition: transform 0.5s cubic-bezier(0.77, 0.2, 0.05, 1.0),
			background 0.5s cubic-bezier(0.77, 0.2, 0.05, 1.0), opacity 0.55s ease;
	}

	.breweries-page a {
		padding-top: 0px;
		margin-left: 0px;
		display: block;
	}
	.breweries-page nav span:first-child {
		transform-origin: 0% 0%;
	}

	.breweries-page nav input:checked ~ .child-2 {
		position: relative;
		top: 15px;
		transform-origin: 0% 100%;
		transform: rotate(-20deg) translate(2px, -1px);
	}

	.breweries-page a.active {
		border-bottom: none;
	}

	.breweries-page input:checked ~ span {
		opacity: 1;
		transform: rotate(45deg) translate(-2px, -1px);
		background: white;
	}

	.breweries-page nav input:checked ~ .child-3 {
		opacity: 0;
		transform: rotate(0deg) scale(0.2, 0.2);
	}

	.breweries-page nav input:checked ~ .child-1 {
		transform: rotate(20deg) translate(0, -1px);
		position: relative;
		top: 15px;
	}

	.breweries-page nav a,
	a.active {
		color: black;
		font-weight: bold;
		margin-left: 0px;
		list-style-type: none;
		-webkit-font-smoothing: antialiased;
		transform-origin: 0% 0%;
		transform: translate(-100%, 0);
		transition: transform 0.5s cubic-bezier(0.77, 0.2, 0.05, 1.0);
		width: 200px;
	}

	.breweries-page nav input:checked ~ a {
		transform: none;
		display: flex;
		position: relative;
		top: 12px;
		left: -7px;
		margin-right: 10px;
		background-color: rgba(226, 240, 227, 0.653);
		border: 1px solid rgba(60, 131, 64, 0.858);
		border-radius: 5px;
		font-size: 15px;
		margin-left: 12px;
		padding: 3px;
		z-index: 4;
	}
	.breweries-page .header-component nav button {
		color: white;
		display: block;
		float: right;
		position: relative;
		top: -111px;
		right: 148px;
		font-weight: bold;
		font-size: 20px;
	}

	.breweries-page nav {
		background-color: rgba(8, 8, 8, 0.774);
		height: 2.3em;
		text-align: left;
	}
}
