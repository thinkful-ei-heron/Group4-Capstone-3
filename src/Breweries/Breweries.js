import React from "react";
import JournalContext from '../contexts/UserContext';
import './Breweries.css';
import Header from '../Header/Header';
import GoogleMapReact from 'google-map-react';
import BarMarker from './Marker/BarMarker'
import BreweryMarker from './Marker/BreweryMarker'
import MapsApiService from '../services/maps-api-service'

class Dashboard extends React.Component {

    static contextType = JournalContext;

    constructor(props) {
        super(props);
        this.state = {
            defaultCenter: {
                lat: 0,
                lng: 0
            },
            center: {
                lat: 0,
                lng: 0
            },
            zoom: 11,
            nearByBars: [],
            nearByBreweries: [],
            value: ''
        }
        this.handleInput = this.handleInput.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
    }

    componentDidMount() {
    }

    handleInput(e) {
        this.setState({value: e.target.value})
    }

    handleSubmit(e) {
        e.preventDefault();
        MapsApiService.getArea(this.state.value).then(res=> {
            let {lat, lng} = res.results;
            console.log(res)
            this.setState({center: {lat, lng}, nearByBars: res.bars, nearByBreweries: res.breweries});
    })
    }

    render() {
        return (
            <>
                <main className='breweries-page'>
                    <Header location={this.props.location} header={'Breweries'}/>
                    <form onSubmit={this.handleSubmit}>
                        <label>Zip Code: <input type='number' onChange={this.handleInput}
                                                value={this.state.value}/></label>
                        <button type='submit'>Go</button>
                    </form>
                    <div style={{height: '100vh', width: '100%'}}>
                        <GoogleMapReact
                            bootstrapURLKeys={{key: 'AIzaSyCG_FMdGssqTRG7tGSvu24UYFopSWQY_-g'}}
                            defaultCenter={this.state.defaultCenter}
                            defaultZoom={this.state.zoom}
                            center={this.state.center}
                        >
                            {this.state.nearByBars.map((place, i)=><BarMarker
                                key={i}
                                lat={place.geometry.location.lat}
                                lng={place.geometry.location.lng}
                                text={place.name}
                                isBeer={false}
                            />)}
                            {this.state.nearByBreweries.map((place, i)=><BreweryMarker
                                key={i}
                                lat={place.geometry.location.lat}
                                lng={place.geometry.location.lng}
                                text={place.name}
                                isBeer={false}
                            />)}
                        </GoogleMapReact>
                    </div>
                </main>
            </>
        )
    }
}

export default Dashboard;