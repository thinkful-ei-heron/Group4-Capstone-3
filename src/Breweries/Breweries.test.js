import React from 'react'
import ReactDOM from 'react-dom'
import { shallow } from 'enzyme'
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json'
import { BrowserRouter } from 'react-router-dom';
import Breweries from './Breweries'


Enzyme.configure({ adapter: new Adapter() });

describe(`Breweries component`, () => {

it('renders without crashing', () => {
  const div = document.createElement('div')
  ReactDOM.render(
    <BrowserRouter>
      <Breweries map = { {list:[], nearByBars:[], nearByBreweries:[], selectedMarker:{name: ''}, }} location={{pathname: ''}} />)
    </BrowserRouter>,
    div)
  ReactDOM.unmountComponentAtNode(div)
})

it('Displays Breweries when rendered', () => {
  const wrapper = shallow(<Breweries map = { {list:[], nearByBars:[], nearByBreweries:[], selectedMarker:{name: ''}, }} location={{pathname: ''}} />)
  expect(toJson(wrapper)).toMatchSnapshot()
})
})