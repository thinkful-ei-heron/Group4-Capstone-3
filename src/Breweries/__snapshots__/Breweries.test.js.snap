// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breweries component Displays Breweries when rendered 1`] = `
<Fragment>
  <main
    className="breweries-page"
    style={
      Object {
        "height": "unset",
      }
    }
  >
    <Header
      header="Breweries"
      location={
        Object {
          "pathname": "",
        }
      }
    />
    <form
      onSubmit={[Function]}
    >
      <label>
        US Zip Code: 
        <input
          onChange={[Function]}
          type="number"
        />
      </label>
      <button
        className="clear-submit-back-btn"
        type="submit"
      >
        Go
      </button>
    </form>
    <div
      className="alert"
      role="alert"
    >
      <p
        className="error"
      />
    </div>
    <KeyComponent />
    <div
      className="breweries-map"
    >
      <GoogleMap
        bootstrapURLKeys={
          Object {
            "key": "AIzaSyCG_FMdGssqTRG7tGSvu24UYFopSWQY_-g",
          }
        }
        debounced={true}
        defaultCenter={
          Object {
            "lat": 40.038629,
            "lng": -105.3716684,
          }
        }
        defaultZoom={11}
        distanceToMouse={[Function]}
        googleMapLoader={[Function]}
        heatmap={Object {}}
        heatmapLibrary={false}
        hoverDistance={30}
        layerTypes={Array []}
        options={[Function]}
        shouldUnregisterMapOnUnmount={true}
        style={
          Object {
            "height": "100%",
            "margin": 0,
            "padding": 0,
            "position": "relative",
            "width": "100%",
          }
        }
        yesIWantToUseGoogleMapApiInternals={false}
      />
    </div>
  </main>
</Fragment>
`;
